src/libNode/FinalBlockProcessing.cpp

```diff
@@ -974,27 +1086,27 @@ bool Node::ProcessFinalBlock(const vector<unsigned char>& message,
             return false;
         }
 
-        if (!CheckStateRoot(txBlock)
-#ifndef IS_LOOKUP_NODE
-            || m_doRejoinAtStateRoot)
+        if (!LOOKUP_NODE_MODE
+            && (!CheckStateRoot(txBlock) || m_doRejoinAtStateRoot))
         {
             RejoinAsNormal();
-#else // IS_LOOKUP_NODE
-        )
+            return false;
+        }
+        else if (LOOKUP_NODE_MODE && !CheckStateRoot(txBlock))
         {
-#endif // IS_LOOKUP_NODE
             return false;
         }
+
         StoreState();
         StoreFinalBlock(txBlock);
 
-#ifndef IS_LOOKUP_NODE
-        BlockStorage::GetBlockStorage().PutMetadata(MetaType::DSINCOMPLETED,
-                                                    {'0'});
-        BlockStorage::GetBlockStorage().PopFrontTxBodyDB();
-#endif // IS_LOOKUP_NODE
+        if (!LOOKUP_NODE_MODE)
+        {
+            BlockStorage::GetBlockStorage().PopFrontTxBodyDB();
```

Comment:
I don't know how this code works, but just wondering why the order of the two lines was flipped.